front/components/agent_builder/settings/TagsSection.tsx

```diff
@@ -0,0 +1,178 @@
+
+async function getTagsSuggestions({
+  owner,
+  instructions,
+  description,
+  tags,
+}: {
+  owner: WorkspaceType;
+  instructions: string;
+  description: string;
+  tags: string[];
+}): Promise<Result<BuilderSuggestionsType, APIError>> {
+  return fetchWithErr(`/api/w/${owner.sId}/assistant/builder/suggestions`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      type: "tags",
+      inputs: {
+        instructions,
+        description,
+        tags,
+        isAdmin: false,
+      },
+    }),
+  });
+}
+
+interface TagsSuggestionsProps {
+  tags: TagType[];
+  onTagsChange: (tags: TagType[]) => void;
+  tagsSuggestions: TagType[];
+}
+
+function TagsSuggestions({
+  tags,
+  onTagsChange,
+  tagsSuggestions,
+}: TagsSuggestionsProps) {
+  const addTag = (tag: TagType) => {
+    const isTagInAgent = tags.findIndex((t) => t.sId === tag.sId) !== -1;
+
+    if (!isTagInAgent) {
+      onTagsChange([...tags, tag]);
+    }
+  };
+
+  return (
+    <div className="flex items-center gap-2">
+      <div className="text-muted-foregroup text-xs font-semibold dark:text-muted-foreground-night">
+        Suggestions:
+      </div>
+
+      {tagsSuggestions.map((tag) => (
+        <Button
+          key={`tag-suggestion-${tag.sId}`}
+          size="xs"
+          variant="outline"
+          label={tag.name}
+          onClick={() => addTag(tag)}
+        />
+      ))}
+    </div>
+  );
+}
+
+export function TagsSection() {
+  const { owner } = useAgentBuilderContext();
+  const { getValues } = useFormContext<AgentBuilderFormData>();
+  const { tags: allTags } = useTags({ owner });
+  const { field } = useController<AgentBuilderFormData, "agentSettings.tags">({
+    name: "agentSettings.tags",
+  });
+  const sendNotification = useSendNotification();
+
+  const selectedTags = field.value;
+
+  const [tagsSuggestions, setTagsSuggestions] =
+    useState<BuilderSuggestionsType>({
+      status: "unavailable",
+      reason: "irrelevant",
+    });
+  const [isTagsSuggestionLoading, setTagsSuggestionsLoading] = useState(false);
+
+  const filteredTagsSuggestions = useMemo(() => {
+    if (tagsSuggestions.status !== "ok") {
+      return [];
+    }
+    const currentTagIds = new Set(selectedTags.map((t) => t.sId));
+    // We make sure we don't suggest tags that already exists.
+    return allTags
+      .filter((t) => !currentTagIds.has(t.sId))
+      .filter((t) => isBuilder(owner) || t.kind !== "protected")
+      .filter(
+        (tag) =>
+          tagsSuggestions.suggestions?.findIndex(
+            (t) => tag.name.toLowerCase() === t.toLowerCase()
+          ) !== -1
+      )
+      .slice(0, 3);
+  }, [tagsSuggestions, selectedTags, allTags, owner]);
+
+  const updateTagsSuggestions = async () => {
+    setTagsSuggestionsLoading(true);
+
+    try {
+      const instructions = getValues("instructions");
+      const description = getValues("agentSettings.description");
```

Comment:
I think Jules mentioned that we use another way to get a value but I forgot what it was 🙈

Reply:
@ykmsd mind adding Max explanation as a short comment ? 🙏🏼

Reply:
perfect, I think we can potentially use this for other fields, I will check this later